libnetwork/portallocator/portallocator.go

```
@@ -248,11 +255,6 @@ func (pm *portMap) getPortRange(portStart, portEnd int) (*portRange, error) {
 		key = pm.defaultRange
 	} else {
 		key = getRangeKey(portStart, portEnd)
-		if portStart == portEnd ||
-			portStart == 0 || portEnd == 0 ||
-			portEnd < portStart {
-			return nil, fmt.Errorf("invalid port range: %s", key)
```

With that change, `getPortRange` doesn't need to return an error anymore.